ingestion/src/metadata/ingestion/source/dashboard/tableau/metadata.py

```diff
@@ -464,3 +440,51 @@ def close(self):
             self.client.sign_out()
         except ConnectionError as err:
             logger.debug(f"Error closing connection - {err}")
+
+    def _get_dashboard(self, dashboard_details: TableauDashboard) -> LineageDashboard:
+        dashboard_fqn = fqn.build(
+            self.metadata,
+            entity_type=LineageDashboard,
+            service_name=self.config.serviceName,
+            dashboard_name=dashboard_details.id,
+        )
+        return self.metadata.get_by_name(
+            entity=LineageDashboard,
+            fqn=dashboard_fqn,
+        )
+
+    def _get_dashboard_datamodel(self, sheet: Sheet) -> DashboardDataModel:
```

Comment:
same here. Maybe pick it up from the context? In looker I am also caching directly in the source some of the models (the explores) so that I don't need to call the API multiple times.